app/scripts/proactive/config.js

```diff
@@ -178,40 +178,89 @@ define(function () {
+import org.ow2.proactive.utils.OperatingSystem;
+import org.ow2.proactive.utils.OperatingSystemFamily;
+
 containerName = "java"
-dockerRunCommand =  "docker run "
-dockerParameters = "--rm --env HOME=/tmp "
+cmd = []
+cmd.add("docker")
+cmd.add("run")
+cmd.add("--rm")
+cmd.add("--env")
+cmd.add("HOME=/tmp")
+
+String osName = System.getProperty("os.name");
+println "Operating system : " + osName;
+OperatingSystem operatingSystem = OperatingSystem.resolveOrError(osName);
+OperatingSystemFamily family = operatingSystem.getFamily();
+
+switch (family) {
+    case OperatingSystemFamily.WINDOWS:
+    	isWindows = true;
+    	break;
+    default:
+        isWindows = false;
+}
+forkEnvironment.setDockerWindowsToLinux(isWindows)
+
 // Prepare ProActive home volume
 paHomeHost = variables.get("PA_SCHEDULER_HOME")
-paHomeContainer = variables.get("PA_SCHEDULER_HOME")
-proActiveHomeVolume = "-v " + paHomeHost + ":" + paHomeContainer + " "
+paHomeContainer = (isWindows ? forkEnvironment.convertToLinuxPath(variables.get("PA_SCHEDULER_HOME")) : variables.get("PA_SCHEDULER_HOME"))
+cmd.add("-v")
+cmd.add(paHomeHost + ":" + paHomeContainer)
 // Prepare working directory (For Dataspaces and serialized task file)
 workspaceHost = localspace
-workspaceContainer = localspace
-workspaceVolume = "-v " + workspaceHost + ":" + workspaceContainer + " "
+workspaceContainer = (isWindows ? forkEnvironment.convertToLinuxPath(workspaceHost) : workspaceHost)
+cmd.add("-v")
+cmd.add(workspaceHost + ":" + workspaceContainer)
+
+cachespaceHost = cachespace
+cachespaceContainer = (isWindows ? forkEnvironment.convertToLinuxPath(cachespaceHost) : cachespaceHost)
+cmd.add("-v")
+cmd.add(cachespaceHost + ":" + cachespaceContainer)
+
+if (!isWindows) {
+     // when not on windows, mount and use the current JRE
+    currentJavaHome = System.getProperty("java.home")
+	forkEnvironment.setJavaHome(currentJavaHome)
+    cmd.add("-v")
+    cmd.add(currentJavaHome + ":" + currentJavaHome)
+}
+
 // Prepare container working directory
-containerWorkingDirectory = "-w " + workspaceContainer + " "
-
-sigar = new org.hyperic.sigar.Sigar()
-try {
-    pid = sigar.getPid()
-    creds = sigar.getProcCred(pid)
-    uid = creds.getUid()
-    gid = creds.getGid()
-    userDefinition = "--user=" + uid + ":" + gid + " "
-} catch (Exception e) {
-    println "Cannot retrieve user or group id : " + e.getMessage()
-    userDefinition = "";
-} finally {
-    sigar.close()
+cmd.add("-w")
+cmd.add(workspaceContainer)
+
+if (isWindows) {
+    // linux on windows does not allow sharing identities (such as AD identities)
+} else {
+    sigar = new org.hyperic.sigar.Sigar()
+    try {
+        pid = sigar.getPid()
+        creds = sigar.getProcCred(pid)
+        uid = creds.getUid()
+        gid = creds.getGid()
+        cmd.add("--user=" + uid + ":" + gid)
+    } catch (Exception e) {
+        println "Cannot retrieve user or group id : " + e.getMessage()
+        userDefinition = "";
```

Comment:
userDefinition is not used, you can remove it

Reply:
right, I removed it